training/src/main/java/com/wolox/training/services/impl/UserServiceImpl.java

```diff
@@ -36,13 +37,13 @@ public void saveUser(User userInput) {
   }
 
   @Override
-  public User findUserById(Long id) throws UserNotFoundException {
+  public User findUserById(@Header("id") Long id) throws UserNotFoundException {
     return this.userRepository.findById(id)
         .orElseThrow(() -> new UserNotFoundException("User not found", 404));
```

Comment:
You will need an onException in the router to catch this UserNotFoundException, to show a 404 HTTPStatus and a message instead of the exception stacktrace